### PR TITLE
packaging: keycloak db backup/restore

### PR DESCRIPTION
**Pull request by arso**
This patch introduces a backup/restore functionality for internally
bundled keycloak database.

Depends on: https://github.com/oVirt/ovirt-engine-keycloak/pull/35

### CODE REVIEW

**Review by didib — COMMENTED** [review]
Generally looks ok. Please verify also restore from a 4.4 backup.

File: packaging/bin/engine-backup.sh.in [hunk 1]
```diff
@@ -2017,6 +2205,12 @@ CINDERLIB_DB_DUMP_FORMAT"
 		eval echo "${var}=\${${var}}"
 	done
 }
+KEYCLOAK_DB_DUMP_COMPRESSOR
```
[the remark this answers is not in the document]
Comment (arso) [reply packaging/bin/engine-backup.sh.in]: Right - good spot!

**Comment by arso** [comment]
OST was good:  https://rhv-devops-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/ds-ost-baremetal_manual/36447/console

**Comment by arso** [comment]
Turning it to draft because I found one incorrect flow. When backup is created from setup without keycloak db and then engine is upgraded to support keycloak (with keycloak setup packages installed) the restore fails because it attempts to restore non-existent keycloak_backup.db from tgz. In such case keycloak db restoration should be skipped.

**Comment by arso** [comment]
I think ost failed becuase there was older version of keycloak (without https://github.com/oVirt/ovirt-engine-keycloak/pull/35)

File: packaging/bin/engine-backup.sh.in [hunk 2]
```diff
@@ -1138,6 +1257,12 @@ dorestore() {
 		generatePgPass
 		verifyConnection "${CINDERLIB_DB_USER}" "${CINDERLIB_DB_HOST}" "${CINDERLIB_DB_PORT}" "${CINDERLIB_DB_DATABASE}"
 	fi
+    if [ -n "${CHANGE_KEYCLOAK_DB_CREDENTIALS}" ]; then
```
Comment (didib) [inline packaging/bin/engine-backup.sh.in:1260]: Please fix the indentation - this file uses TABs. Sadly github does not show this...

Comment (arso) [reply packaging/bin/engine-backup.sh.in]: fixed.

File: packaging/bin/engine-backup.sh.in [hunk 3]
```diff
@@ -1958,6 +2143,9 @@ ${DWH_DB_HOST}:${DWH_DB_PORT}:${DWH_DB_DATABASE}:${DWH_DB_USER}:${dwh_password}
 __EOF__
 	[ -n "${CINDERLIB_DB_USER}" ] && cat >> "${MYPGPASS}" << __EOF__
 ${CINDERLIB_DB_HOST}:${CINDERLIB_DB_PORT}:${CINDERLIB_DB_DATABASE}:${CINDERLIB_DB_USER}:${cinderlib_password}
+__EOF__
+    [ -n "${KEYCLOAK_DB_USER}" ] && cat >> "${MYPGPASS}" << __EOF__
```
[the remark this answers is not in the document]
Comment (arso) [reply packaging/bin/engine-backup.sh.in]: fixed

**Review by didib — APPROVED** [review]

**Review by mwperina — APPROVED** [review]
+1

**Pull request merged**